### DISCUSSION TITLE
idea: use offline maps from osmand

### DISCUSSION THREAD

**Discussion by alexmo1997** [owntracks/android#1243] [Ideas & New Features]
I'm not entirely sure this is possible, but I think a lot of people already use osmand and thus have offline maps (for osm) available. If owntracks could make use of this, owntracks could have offline support without having to use additional space if you already have the maps downloaded.

**Comment by growse** [comment]
I'm not sure OSMand make any API available that would allow anyone else to get at their local maps storage. Android apps are isolated from each other by design, so unless there's a specific mechanism this is likely to be very difficult to impossible.

Could use some research though.

**Reply by growse** [reply]
I did some research!

Ok, so OsmAnd have an API and an SDK, that seems to allow other apps to embed an OsmAnd map in their app, as long as you've got their app installed as well.

It totally hadn't occurred to me that this was a mechanism that they provided, but given that they do, it seems reasonable that we should support an OsmAnd-rendered layer if possible. Will do some more research and see how feasible it looks.

**Reply by alexmo1997** [reply]
That's really cool, it's pretty much exactly what I had hoped for. Thank you for looking into it, that's great to hear! 🎉

**Reply by jmencak** [reply]
First of all, thank you for a great project. I've set this up over the weekend and I love it.

> If owntracks could make use of this, owntracks could have offline support without having to use additional space

I just wanted to add this would not only be useful for saving additional space, but also making it possible to use the app for people with very limited monthly data plans.
